communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java

```diff
@@ -61,7 +59,12 @@
 
     @Override
     public Optional<GAVDependencyTree> getDependencyTreeOfGAV(GAV gav)
-            throws CommunicationException {
+            throws CommunicationException, FindGAVDependencyException {
+
+        if (!doesGAVExistInPublicRepo(gav)) {
+            throw new FindGAVDependencyException("Could not find: " + gav
+                    + "in public repo of Aprox");
```

Comment:
I agree with this approach to throw an exception in this case. It is an exceptional state.